### PR TITLE
variables : add $(LABELS_ICONS) to use colorlabels in patterns

### PR DESCRIPTION
**Pull request by AlicVB**
only for info patterns, not for export due to markup contents

### CODE REVIEW

**Comment by johnny-bit** [comment]
Would changing this for export pattern to color name be better? that way info's there just not supper pretty. Eg:
`<span foreground=\"#ee0000\">⚫ </span>` -> `(red ⚫)`?

that way if somebody uses `$(LABELS_ICONS)` instead of `$(LABELS)` the text won't be empty?

or maybe in case of export fall `$(LABELS_ICONS)` to `$(LABELS)`?
 (meaning: turn `else if(has_prefix(variable, "LABELS"))` into `else if(has_prefix(variable, "LABELS") || has_prefix(variable, "LABELS_ICONS"))` since previous condition should handle case of infos jobcode?

**Comment by AlicVB** [comment]
@johnny-bit : good idea... done.
So now in case we use $(LABELS_ICONS) in export pattern, we fall back to $(LABELS)

File: src/common/variables.c [hunk 1]
```diff
@@ -371,7 +372,46 @@ static char *get_base_value(dt_variables_params_t *params, char **variable)
...
-  else if(has_prefix(variable, "LABELS"))
+  else if(has_prefix(variable, "LABELS_ICONS") && g_strcmp0(params->jobcode, "infos") == 0)
+  {
+    escape = FALSE;
+    GList *res = dt_metadata_get(params->imgid, "Xmp.darktable.colorlabels", NULL);
+    res = g_list_first(res);
+    gchar *txt = NULL;
+    if(res != NULL)
+    {
+      GList *labels = NULL;
```
Comment (TurboGit) [inline src/common/variables.c:383]: can be removed as not used.

File: src/common/variables.c [hunk 2]
```diff
@@ -371,7 +372,46 @@ static char *get_base_value(dt_variables_params_t *params, char **variable)
...
+        {
+          txt = dt_util_dstrcat(txt, "<span foreground=\"#0000ee\">⚫ </span>");
+        }
+        else if(g_strcmp0(lb, "purple") == 0)
+        {
+          txt = dt_util_dstrcat(txt, "<span foreground=\"#ee00ee\">⚫ </span>");
+        }
+      } while((res = g_list_next(res)) != NULL);
+      result = g_strdup(txt);
+      g_list_free(labels);
```
Comment (TurboGit) [inline src/common/variables.c:409]: likewise

File: src/common/variables.c (hunk 1 above)
Comment (AlicVB) [reply src/common/variables.c]: you're right...

File: src/common/variables.c (hunk 2 above)
Comment (AlicVB) [reply src/common/variables.c]: done

**Comment by TurboGit** [comment]
All good. Thanks.

**Comment by GLLM** [comment]
Hi !
I've tested this in Darkroom infoline ( in dt 3.1.0~git2270.f42aeca9a on Linux Kubuntu 20.04) and I see this
image
i.e. the dots are not in color.
Is this expected ?
Thanks
GLLM

**Comment by johnny-bit** [comment]
Huh, interesting - I can confirm @GLLM's finding! `$(LABELS_ICONS)` doesn't work at all for me :/ @GLLM - can you open an issue for it?

**Comment by GLLM** [comment]
Sure, will do it right away ! :)

**Pull request merged**